frontend/src/routes/manage/Video/Shared.tsx

```diff
@@ -0,0 +1,197 @@
+        const params = regex.exec(url.pathname);
+        if (params === null) {
+            return null;
+        }
+
+        const videoId = decodeURIComponent(params[1]);
+        const queryRef = loadQuery<SharedVideoManageQuery>(query, { id: `ev${videoId}` });
+
+        return {
+            render: () => <RootLoader
+                {...{ query, queryRef }}
+                noindex
+                nav={data => [
+                    <BackLink key={1} />,
+                    <ManageVideoNav key={2} event={data.event} active={page} />,
+                ]}
+                render={data => {
+                    if (data.event === null) {
+                        return <NotFound kind="video" />;
+                    }
+                    if (data.event.__typename !== "AuthorizedEvent" || !data.event.canWrite) {
+                        return <NotAuthorized />;
+                    }
+
+                    return render(data.event);
+                }}
+            />,
+            dispose: () => queryRef.dispose(),
+        };
+    })
+);
+
+
+// We have one query for all "manage video" pages as there is a huge overlap in
+// what they request. It just simplifies our code a lot and we only pay by
+// overfetching a bit.
+const query = graphql`
+    query SharedVideoManageQuery($id: ID!) {
+        ...UserData
+        event: eventById(id: $id) {
+            __typename
+            ... on AuthorizedEvent {
+                id
+                title
+                description
+                opencastId
+                created
+                canWrite
+                isLive
+                syncedData {
+                    duration
+                    thumbnail
+                    updated
+                    startTime
+                    endTime
+                    tracks { flavor resolution mimetype }
+                }
+                series {
+                    title
+                    opencastId
+                    ...SeriesBlockSeriesData
+                }
+                hostRealms { id isRoot name path }
+            }
+        }
+    }
+`;
+
+
+/** Simple nav element linking back to "my videos" overview page. */
+const BackLink: React.FC = () => {
+    const { t } = useTranslation();
+    const items = [
+        <LinkWithIcon key={MANAGE_VIDEOS_PATH} to={MANAGE_VIDEOS_PATH} iconPos="left">
+            <FiCornerLeftUp />
+            {t("manage.nav.my-videos")}
+        </LinkWithIcon>,
+    ];
+
+    return <LinkList items={items} />;
+};
+
+
+type ManageVideoNavProps = {
+    event: Event;
+    active: ManageVideoSubPageType;
+};
+
+const ManageVideoNav: React.FC<ManageVideoNavProps> = ({ event, active }) => {
+    const { t } = useTranslation();
+
+    if (event === null) {
+        return null;
+    }
+    if (event.__typename !== "AuthorizedEvent" || !event.canWrite) {
+        return null;
+    }
+
+    const id = event.id.substring(2);
```

Comment:
Any reason to use `substring` here instead of `slice` like everywhere else? 🤔

Reply:
Nop. At least in that file I don't use `slice`.

Reply:
OK, `substring` is used in one other place as well so no need to get nitpicky here. :shrug: